File: scripts/Aulas/Pré-processamento de dados/Python/logical-subsetting.py
# LOGICAL SUBSETTING

import pandas # Importa a biblioteca Pandas para ser utilizada no código

vector1 = [188.2, 181.3, 193.4, 192.3] # Cria um vetor dinâmico com tamanho variado
vector2 = ["Jeff", "Roger", "Andre", "Brian"] # Cria um vetor dinâmico com tamanho variado
data = {'Heights': vector1, 'First_names': vector2} # Cria uma variável que armazena dois títulos com os dois vetores como valores
myDataFrame = pandas.DataFrame(data) # Cria um data frame usando a variável criada anteriormente como valor, listando de forma ordenada os dois valores (vetores)

is_jeff = myDataFrame.First_names == "Jeff" # Cria uma variável para servir de função comparativa, ou seja, apenas aqueles que tem o "First_names" "Jeff" poderá aparecer na consulta ao data frame utilizando esta variável
small_heights = myDataFrame.Heights < 190 # Cria uma variável para servir de função comparativa, ou seja, apenas aqueles que tem o "Heights" menor que 190 poderá aparecer na consulta ao data frame utilizando esta variável

print(myDataFrame.loc[is_jeff]) # Printa o valor do data frame que corresponda a variável de comparação
print("\n=======================\n") # Printa um separador
print(myDataFrame[small_heights]) # Printa o valor do data frame que corresponda a variável de comparação

File: scripts/Aulas/Pré-processamento de dados/Julia/logical-subsetting.jl
# LOGICAL SUBSETTING

using DataFrames # Importa a biblioteca DataFrames para ser utilizada no código

vector1 = [188.2, 181.3, 193.4, 192.3] # Cria um vetor dinâmico com tamanho variado
vector2 = ["Jeff", "Roger", "Andre", "Brian"] # Cria um vetor dinâmico com tamanho variado
myDataFrame = DataFrame(Heights = vector1, First_names = vector2) # Cria um data frame usando os dois vetores criados anteriormente, atribuindo o nomes as colunas em que os dois serão posicionados

print(myDataFrame[myDataFrame.First_names .== "Jeff", :]) # Printa o valor do data frame que corresponda a todos os "First_names" iguais a "Jeff"
print("\n\n===============\n\n") # Printa um separador
print(myDataFrame[myDataFrame.Heights .< 190, :]) # Printa o valor do data frame que corresponda a todos os "Heights" menor que 190